src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -209,3 +214,37 @@ fn make_data(book: &MDBook) -> Result<serde_json::Map<String, serde_json::Value>
     debug!("[*]: JSON constructed");
     Ok(data)
 }
+
+fn build_header_links(mut html: String) -> String {
+    for header in &["h1", "h2", "h3", "h4", "h5"] {
+        let regex = Regex::new(&format!("<{h}>(.*?)</{h}>", h=header)).unwrap();
+
+        html = regex.replace_all(&html, |caps: &Captures| {
+            let text = &caps[1];
+            let mut id = text.to_string();
```

Comment:
i am almost tempted to pull this shenanigans out into a microcrate so that rustdoc and mdbook can share it